src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Unix.cs

```diff
@@ -16,19 +13,29 @@ namespace System.IO.Pipes
     /// </summary>
     public sealed partial class NamedPipeClientStream : PipeStream
     {
-        private static string GetPipePath(string serverName, string pipeName)
-        {
-            throw NotImplemented.ByDesign; // TODO: Implement this
-        }
-
-        // Waits for a pipe instance to become available. This method may return before WaitForConnection is called
-        // on the server end, but WaitForConnection will not return until we have returned.  Any data writen to the
-        // pipe by us after we have connected but before the server has called WaitForConnection will be available
-        // to the server after it calls WaitForConnection. 
         [SecurityCritical]
-        private void ConnectInternal(int timeout, CancellationToken cancellationToken, int startTime)
+        private bool TryConnect(int timeout, CancellationToken cancellationToken)
         {
-            throw NotImplemented.ByDesign; // TODO: Implement this
+            try
+            {
+                // Open the file.  For In or Out, this will block until a client has connected.
+                // Unfortunately for InOut it won't; this is different from the Windows behavior;
+                // it won't block for InOut until it actually performs a read or write operation.
```

Comment:
It's hard to know at a glance to which OS this last line refers. 

First time I've seen a double `;` ;)